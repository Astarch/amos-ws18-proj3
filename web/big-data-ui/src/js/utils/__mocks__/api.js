/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/

const auth = {
  postLogin: jest.fn(() => Promise.resolve({success: true})),
  postRegistration: jest.fn(() => Promise.resolve({success: true, response:{status: 401}})),
  postCompleteRegistration: jest.fn(() => Promise.resolve({success: true})),
  doLogout: jest.fn(() => Promise.resolve({success: true}))
};

/***** user *****/

const user = {
  getAll: jest.fn(() => Promise.resolve({success: true}))
};

/***** graph *****/

const graph = {
  getData: jest.fn(() => Promise.resolve({success: true}))
};

const api = {
  auth: auth,
  user: user,
  graph: graph
}

export  {
  api
};

