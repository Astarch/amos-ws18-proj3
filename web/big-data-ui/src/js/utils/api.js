import axios from 'axios';
import qs from 'qs';
import store from 'store2'


/******************** Base API / Client Configuration ********************/

let port = "8081/"
let localhostUrl = "http://localhost:" + port;
let stagingUrl = "http://18.216.129.153:" + port;
let liveUrl = "http://18.216.122.218:" + port;
let baseUrl = stagingUrl
baseUrl = (GIT.BRANCH && !GIT.BRANCH.includes("master")) ? liveUrl : stagingUrl;



function writeAuthToken(token) {
    store.session('authToken', token)
    return token
}

function getAuthToken() {
    return store.session('authToken')
}


const axiosInstance = axios.create({
    baseURL: baseUrl,
});

if (getAuthToken() && getAuthToken().length > 5) {
    Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': getAuthToken() } });
}

// Intercepts all responses, retrieves the x auth token and sets it to all following requests headers!
axiosInstance.interceptors.response.use(response => {
    let authToken = getAuthToken()
    if (response.status == 401) {
        authToken = writeAuthToken("")
    } else if (response && response.headers && response.headers['x-auth-token']) {
        let authToken = writeAuthToken(response.headers['x-auth-token']);
        if (authToken && authToken.length > 5) {
            Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': authToken } });
        }
    }

    return response;
}, error => {
    // Do something with response error
    return Promise.reject(error);
});

export default axiosInstance;

export const isLoggedIn = () => {
    let authToken = getAuthToken() ? getAuthToken() : authToken;
    return authToken && authToken.length > 5
}

export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getData = (path) => axiosInstance.get(path);

const graph = {
    getData
};

const api = {
    auth,
    user,
    graph
}

export {
    api
};