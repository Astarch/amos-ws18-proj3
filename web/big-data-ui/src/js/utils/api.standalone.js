import axios from 'axios';
import qs from 'qs';
import storage from 'store2'
import store from 'src/js/store';

import trump from 'src/assets/wordcounts/trump'
const availableTerms = ["trump"]



/******************** Base API / Client Configuration ********************/

function writeAuthToken(token) {
  storage.set('authToken', token)
  return token
}

function getAuthToken() {
  return storage.get('authToken')
}

export const isLoggedIn = () => {
  let authToken = getAuthToken() ? getAuthToken() : authToken;
  return authToken && authToken.length > 5
}

export const logoutUser = () => {
  writeAuthToken("")
}


/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => {
  return new Promise((resolve, reject) => {
    if (username == "user2" && password == "pass2") {
      const response = {
        data: {
          id: 123,
          username: "user2",
          email: "user2@test.de",
          firstname: "John",
          lastname: "Doee",
          authorities: [],
          isLoggedIn: true
        }
      }
      writeAuthToken("loggedIn")
      resolve(response)
    } else {
      const errorResponse = {
        status: 401
      }
      reject(errorResponse)
    }
  })
}
//axiosInstance.post('/auth/login', qs.stringify({
//    username: username,
//    password: password
//}));

const postCompleteRegistration = (username,
  password,
  firstname = username,
  lastname = username,
  email = username + "@test.com",
  address = "addres1",
  phone = "phoneNr") => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "password": password,
//    "firstname": firstname,
//    "lastname": lastname,
//    "email": email,
//    "address": address,
//    "phone": phone
//});

const postRegistration = (username, firstname, lastname,
  email,
  password) => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "firstname": firstname,
//    "lastname": lastname,
//    "password": password,
//    "email": email,
//});



const auth = {
  postLogin,
  postRegistration,
  postCompleteRegistration
};

const localFirebaseUrl = "http://localhost:5000/pretrendr-d5939/us-central1"
const remoteFirebaseUrl = "https://us-central1-pretrendr-d5939.cloudfunctions.net"
const firebaseUrl = remoteFirebaseUrl
/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) => {
  /*return new Promise((resolve, reject)=> {
      if(query.indexOf("trump")>= 0){
          const response = {
              data: trump
          }
          resolve(response)
      }else{
          const errorResponse = {
              status: 404
          }
          reject(errorResponse)
      }
  })*/
  return axios.get(`${firebaseUrl}/getWordcountData?query=${query}&method=${method}`)

  //axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);
}
const uploadDataToFirestore = (query, method, timerange, data) => {
  const uploadData = {
    query,
    method,
    timerange,
    data,
  };
  return axios.post(`${firebaseUrl}/addWordcountData`, uploadData)    
}

const getCachedTrends = () => axios.get(`${firebaseUrl}/getAllCachedTerms`)


const graph = {
  getWordcountByDay,
  getCachedTrends
};

const api = {
  auth,
  graph
}

export {
  api
};
