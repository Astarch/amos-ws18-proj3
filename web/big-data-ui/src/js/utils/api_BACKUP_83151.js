import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

<<<<<<< HEAD
export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username, firstname,lastname,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "firstname": firstname,
    "lastname": lastname,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) =>
    axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);

const graph = {
    getWordcountByDay
};

const uploadDataToFirestore = (query, method, timerange, data) => {
    const uploadData = {
      query,
      method,
      timerange,
      data,
    };
    return axios.post('https://us-central1-pretrendr-d5939.cloudfunctions.net/addWordcountData', uploadData)    
  }

const api = {
    auth,
    user,
    graph,
    uploadDataToFirestore
}
=======
const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi
>>>>>>> initial commit to use frontend without backend

export {
    isLoggedIn,
    logoutUser,
    api
}
