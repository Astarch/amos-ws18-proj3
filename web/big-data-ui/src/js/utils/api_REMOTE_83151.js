import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi

export {
    isLoggedIn,
    logoutUser,
    api
}
