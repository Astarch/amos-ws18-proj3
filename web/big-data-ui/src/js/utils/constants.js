const FormTypeEnum = {"login": 1, "register": 2};
Object.freeze(FormTypeEnum);

export {
  FormTypeEnum
}
