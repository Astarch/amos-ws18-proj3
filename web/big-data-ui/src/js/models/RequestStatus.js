import { ServerErrors } from './../utils/constants';

export default class RequestStatus {
    constructor(name = '', success = false, failure = false, pending = false, failureType = ServerErrors.ERROR_NONE) {
        this.name = name;
        this.success = success;
        this.failure = failure;
        this.pending = pending;
        this.failureType = failureType;
    }

    toString() {
        return `RequestStatus(${this.name})[ success:${this.success}, failure:${this.failure}, pending:${this.pending}, failureType:${this.failureType} ]`
    }
}