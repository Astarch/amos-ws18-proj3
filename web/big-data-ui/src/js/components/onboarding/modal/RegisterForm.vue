<template>
  <div id="form-register" class="onboarding-form form-register" :class="{active: active}">
    <input id="register-username" type="text" name="name" placeholder="Username" v-model="username" v-on:keyup.enter.stop="submit()" v-on:input="hideError">
    <input id="register-firstname" type="text" name="firstname" placeholder="Firstname" v-model="firstname" v-on:keyup.enter.stop="submit()" v-on:input="hideError">
    <input id="register-lastname" type="text" name="lastname" placeholder="Lastname" v-model="lastname" v-on:keyup.enter.stop="submit()" v-on:input="hideError">
    <input id="register-email" type="email" name="email" placeholder="Email" v-model="email" v-on:keyup.enter.stop="submit()" v-on:input="checkEmail" ref="emailInput" v-bind:style="regEmailStyle">
    <input id="register-password" type="password" name="password" placeholder="Password" v-model="password" v-on:keyup.enter.stop="submit()" v-on:input="checkPassword" ref="pwinput" v-bind:style="regPwStyle">
    <div v-if="hasError" class="alert alert-danger" role="alert">
      {{errorText}}
    </div>
    <input id="registerSubmit" type="submit" :class="{disabled: requestStatus.pending}" v-on:click.stop="submit()" value="Register">
    <div v-if="requestStatus.success && !requestStatus.pending" class="alert alert-success" role="alert">
      Congratulations {{user.username}}! You can now start to use pretrendr.
    </div>
    <div class="links">
      <a href="" v-on:click.prevent.stop="openLogin()">Already have an account?</a>
    </div>
  </div>
</template>

<script>
  import {
    mapState
  } from "vuex";
  import {
    isPasswordValid,
    isNameValid,
    isMailValid,
    getPasswordStrength
  } from "../../../utils/validation";
  import RequestStatus from 'src/js/models/RequestStatus';
  import {
    ServerErrors
  } from 'src/js/utils/constants';
  
  const colors = [
    "",
    "#F09090",
    "#EBA096",
    "#E5B09C",
    "#E0C0A2",
    "#DBCFA7",
    "#D6DFAD",
    "#D0EFB3",
    "#CBFFB9"
  ];
  
  export default {
    name: "register-form",
    props: {
      active: Boolean
    },
    data: () => ({
      username: "",
      firstname: "",
      lastname: "",
      email: "",
      password: "",
      errorText: "",
      requestStatus: new RequestStatus(),
      hasError: false,
  
      // Style Password Input Field
      regPwStyle: {
        backgroundColor: ""
      },
  
      // Style Email Input Field
      regEmailStyle: {
        backgroundColor: ""
      }
    }),
    computed: mapState({
      // arrow functions can make the code very succinct!
      user: state => state.user.user
    }),
    mounted() {
      this.clearRegistration()
    },
    methods: {
      submit() {
        //this.isSubmitting = true;
  
        let validData = this.validateRegistration(
          this.username,
          this.firstname,
          this.lastname,
          this.email,
          this.password
        );
  
        if (validData) {
          this.doRegistration(this.username, this.firstname, this.lastname, this.email, this.password);
        }
      },
      doRegistration(name, firstname ,lastname,  email, password) {
        this.requestStatus.pending = true;
        this.$store
          .dispatch("registerUser", {
            username: name,
            firstname: firstname,
            lastname: lastname,
            email,
            password
          })
          .then(requestStatus => {
            this.requestStatus = requestStatus;
            this.openLogin();
          })
          .catch(error => {

            this.requestStatus = error;
            if (this.requestStatus !== ServerErrors.ERROR_SERVER) {
              this.showError("User already registered!");
            } else {
              this.showError("Server error - please try again later!");
            }
          });
      },
      clearRegistration() {
        this.username = "";
        this.firstname = "";
        this.lastname = "";
        this.email = "";
        this.password = "";
        this.requestStatus = new RequestStatus();
      },
      validateRegistration(name, firstname, lastname, mail, password) {
        let validRegistration = false;
        if (!isNameValid(name)) {
          this.showError("Please enter a valid name!");
        } else if (!isNameValid(firstname)) {
          this.showError("Please enter a valid firstname!");
        } else if (!isNameValid(lastname)) {
          this.showError("Please enter a valid lastname!");
        } else if (!isMailValid(mail)) {
          this.showError("Please enter a valid mail!");
        } else if (!isPasswordValid(password)) {
          this.showError("Please enter a valid password!");
        } else {
          validRegistration = true;
          this.hideError();
        }
        return validRegistration;
      },
      checkEmail: function() {
        this.hideError();
        if (this.email.length > 0) {
          if (isMailValid(this.email)) {
            this.regEmailStyle.backgroundColor = colors[colors.length - 1];
          } else {
            this.regEmailStyle.backgroundColor = colors[1];
          }
        } else {
          this.regEmailStyle.backgroundColor = colors[0];
        }
      },
      checkPassword: function() {
        this.hideError();
        var strength = getPasswordStrength(this.password);
  
        if (strength > 0 && strength < colors.length) {
          this.regPwStyle.backgroundColor = colors[strength];
        } else {
          this.regPwStyle.backgroundColor = colors[0];
        }
      },
      showError(message) {
        if (typeof message === "string" || message instanceof String) {
          this.hasError = true;
          this.errorText = message;
        } else {
          this.hasError = false;
        }
      },
      hideError() {
        this.hasError = false;
      },
      openLogin() {
        this.$emit("openLogin");
      }
    }
  };
</script>

<style lang='scss' scoped>
  @import "../../../../scss/forms";
  div.alert-success {
    margin-top: 10px;
  }
</style>
