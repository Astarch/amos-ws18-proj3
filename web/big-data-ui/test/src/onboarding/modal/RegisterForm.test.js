import { shallow, mount } from 'vue-test-utils'
import Vue from 'vue'
import Vuex from 'vuex'

import RegisterForm from 'src/js/components/onboarding/modal/RegisterForm'
import RequestStatus from 'src/js/models/RequestStatus';
import { api } from 'src/js/utils/api'
jest.mock('src/js/utils/api');

Vue.use(Vuex)

describe('RegisterForm.test.js', () => {

    let store
    let actions
    let state

    let registerMock

    beforeEach(() => {
        jest.resetModules()
        jest.clearAllMocks()

        let reqStatus = Object.assign({}, new RequestStatus(), {
            pending: false,
            success: true,
        })
        registerMock = jest.fn()
        registerMock
            .mockReturnValue(reqStatus)

        state = {
            user: {
                user: {
                    id: '',
                    username: '',
                    email: '',
                    encryptedPassword: '',
                    firstname: '',
                    lastname: '',
                    authorities: []
                }
            }
        }
        actions = {
            registerUser: registerMock
        }
        store = new Vuex.Store({
            state,
            actions
        })

    })

    test('has the expected html structure when not active', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.setProps({ active: false })
        expect(cmp.element).toMatchSnapshot('inactive')
    })

    test('has the expected html structure when active', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.setProps({ active: true })
        expect(cmp.element).toMatchSnapshot('active')
    })

    test('to login clicked event is emitted', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.find('div.links > a').trigger('click')
        expect(cmp.emitted().openLogin).toBeTruthy()
    })


    test('error is shown when submitting empty inputs', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.find('#registerSubmit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })


    let validPassword = "test1234"
    let invalidPassword = ""
    let validUsername = "test"
    let invalidUsername = ""
    let validEmail = "test@test.de"
    let invalidEmail = "testtest.de"

    test('error is shown when submitting empty password', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.find('input#register-username').element.value = validUsername
        cmp.find('input#register-username').trigger('input')
        cmp.find('input#register-email').element.value = validEmail
        cmp.find('input#register-email').trigger('input')
        cmp.find('#registerSubmit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('error is shown when submitting empty username', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.find('input#register-password').element.value = validPassword
        cmp.find('input#register-password').trigger('input')
        cmp.find('input#register-email').element.value = validEmail
        cmp.find('input#register-email').trigger('input')
        cmp.find('#registerSubmit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('error is shown when submitting empty mail', () => {
        const cmp = shallow(RegisterForm, { store })
        cmp.find('input#register-password').element.value = validPassword
        cmp.find('input#register-password').trigger('input')
        cmp.find('input#register-username').element.value = validUsername
        cmp.find('input#register-username').trigger('input')
        cmp.find('#registerSubmit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('correct data is submitted', () => {
        const cmp = shallow(RegisterForm, { store })
        let email = validEmail
        let username = validUsername
        let password = validPassword
        cmp.find('input#register-password').element.value = password
        cmp.find('input#register-password').trigger('input')
        cmp.find('input#register-username').element.value = username
        cmp.find('input#register-username').trigger('input')
        cmp.find('input#register-email').element.value = email
        cmp.find('input#register-email').trigger('input')
        cmp.find('#registerSubmit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeFalsy()

        expect(actions.registerUser.mock.calls).toHaveLength(1)
        expect(actions.registerUser.mock.calls[0][1]).toEqual({ username, email, password })
    })

})