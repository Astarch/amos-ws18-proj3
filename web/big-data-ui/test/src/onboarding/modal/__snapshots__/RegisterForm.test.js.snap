// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm.test.js has the expected html structure when active: active 1`] = `
<div
  class="onboarding-form form-register active"
  id="form-register"
>
  <input
    id="register-username"
    name="name"
    placeholder="Name"
    type="text"
  />
   
  <input
    id="register-email"
    name="email"
    placeholder="Email"
    type="email"
  />
   
  <input
    id="register-password"
    name="password"
    placeholder="Password"
    type="password"
  />
   
  <!---->
   
  <input
    class=""
    id="registerSubmit"
    type="submit"
    value="Register"
  />
   
  <!---->
   
  <div
    class="links"
  >
    <a
      href=""
    >
      Already have an account?
    </a>
  </div>
</div>
`;

exports[`RegisterForm.test.js has the expected html structure when not active: inactive 1`] = `
<div
  class="onboarding-form form-register"
  id="form-register"
>
  <input
    id="register-username"
    name="name"
    placeholder="Name"
    type="text"
  />
   
  <input
    id="register-email"
    name="email"
    placeholder="Email"
    type="email"
  />
   
  <input
    id="register-password"
    name="password"
    placeholder="Password"
    type="password"
  />
   
  <!---->
   
  <input
    class=""
    id="registerSubmit"
    type="submit"
    value="Register"
  />
   
  <!---->
   
  <div
    class="links"
  >
    <a
      href=""
    >
      Already have an account?
    </a>
  </div>
</div>
`;
